extra_data/components.py

```diff
@@ -864,6 +863,118 @@ def _make_image_index(self, tids, inner_ids, inner_name='pulse'):
         )
 
 
+class JUNGFRAU(MPxDetectorBase):
+    """An interface to JUNGFRAU data.
+
+    Parameters
+    ----------
+
+    data: DataCollection
+      A data collection, e.g. from RunDirectory.
+    modules: set of ints, optional
+      Detector module numbers to use. By default, all available modules
+      are used.
+    detector_name: str, optional
+      Name of a detector, e.g. 'SPB_IRDA_JNGFR'. This is only needed
+      if the dataset includes more than one JUNGFRAU detector.
+    min_modules: int
+      Include trains where at least n modules have data. Default is 1.
+    """
+    # We appear to have a couple of different formats for source names:
+    # SPB_IRDA_JNGFR/DET/MODULE_1:daqOutput  (e.g. in p 2566, r 61)
+    # SPB_IRDA_JF4M/DET/JNGFR03:daqOutput    (e.g. in p 2732, r 12)
+    # This should catch both. We may need to generalise it more in the future.
+    _source_re = re.compile(
+        r'(?P<detname>.+_(JNGFR|JF4M))/DET/(MODULE_|JNGFR)(?P<modno>\d+)'
```

Comment:
Last time I checked, these 2 other naming form existed (hopefully there are no additional in the meantime :fearful: )
`FXE_XAD_JF1M/DET/RECEIVER-1`
`HED_IA1_JNGFR-1/DET/MODULE`

Reply:
Thanks, I'll add those to the regex. Do you have proposal & run numbers handy containing those patterns?

Reply:
Hmm, if it's really `HED_IA1_JNGFR-1/DET/MODULE` that might not work - this expects to find a module number. But if it's just one module, you don't really need this interface anyway, because you can do e.g. `run['HED_IA1_JNGFR-1/DET/MODULE', 'data.adc'].xarray()`.

Reply:
`HED_IA1_JNGFR-1/DET/MODULE`
`HED_IA1_JNGFR-2/DET/MODULE`
...
?

:upside_down_face:

Reply:
But I am also not certain that HED is using JF in a aggregated way :thinking:
I think they will be used together (if you remember the publication with the projected plan), but they won't be assembled in a case

Reply:
This would explain the naming - they are considering these as completely independent detectors, single-module...

Reply:
OK, let's leave the HED naming out for now, and find a way to make it work if we need to. I assume that the FXE naming has modules `RECEIVER-1`, `RECEIVER-2`, etc.?

Reply:
Looks like all JF devices have been consistently renamed :relieved: 
I interpret that the `500K` are independent detectors, while others are multimodules

# FXE

FXE_XAD_JF1M/DET/JNGFR01
FXE_XAD_JF1M/DET/JNGFR02

FXE_XAD_JF500K/DET/JNGFR03

# HED

HED_IA1_JF500K1/DET/JNGFR01
HED_IA1_JF500K2/DET/JNGFR02
HED_IA1_JF500K3/DET/JNGFR03
HED_IA1_JF500K4/DET/JNGFR04

# SPB

SPB_IRDA_JF4M/DET/JNGFR01
SPB_IRDA_JF4M/DET/JNGFR02
...
SPB_IRDA_JF4M/DET/JNGFR08

Reply:
As far as I know, SPB will stick to their assembly and naming (no further changes to be expected). And I think it's easier to deal with the HED naming now, as they number not only the JF500KN as independent detectors, but have a number in the terminal module segment as well.

Reply:
Yes, we can work with the `HED_IA1_JF500K2/DET/JNGFR02` style names, in case that's useful.

Reply:
I just realised that there's a question to for names like `HED_IA1_JF500K1/DET/JNGFR01`. The code tries to identify and check a 'detector name'. The obvious way to do that is to have `HED_IA1_JF500K1` as the name, but then it won't let you combine that data with `HED_IA1_JF500K2`, because that's a separate instance. We could make `HED_IA1_JF500K` the name, so it can pull together data from all of them in a single array - but that might be confusing if people do see them as separate detectors. Or we can leave the 500K names out of this for now (you can still use them as individual sources, of course).

Thoughts?

Reply:
> Or we can leave the 500K names out of this for now

hum, I lean in this direction. Seems clear that `500K` devices are meant to be used as standalone detectors. And this class represent mutimodular detectors (If we do this, then maybe emphasize in the docstring of the JUNGFRAU object that it is mean to abstract multimodular jungfrau detectors only). Unless there are use cases where it makes sens to handle and combine data from multiple detector (non modular) but we can review this once the case comes up.